core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -169,8 +168,7 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     waitForTopics(client, expectedPresent = topics, expectedMissing = List())
 
     val controller = brokers.find(_.config.brokerId == brokers.flatMap(_.metadataCache.getControllerId).head).get
-    controller.shutdown()
-    controller.awaitShutdown()
+    killBroker(controller.config.brokerId)
```

Comment:
The point of this test is a little unclear to me. We have already verified metadata propagation in the `waitForTopics` above. Not sure why killing the controller would change that. Anyway, perhaps a more accurate name for the test would be something like `testCreatedTopicStillVisibleAfterControllerChange`? I feel like we could also just delete the test since it is probably covered by every integration test which kills the controller.

Reply:
> We have already verified metadata propagation in the waitForTopics above

I think this assumption is not correct. `waitForTopics` sends a `metadata` API call to a broker (not necessarily the controller). In such a situation, it is not guaranteed that the broker will have the latest metadata in the metadata cache. (e.g. maybe the broker listener has not updated it's metadata cache from zk yet). Am I right in saying this?

>  I feel like we could also just delete the test since it is probably covered by every integration test which kills the controller.

The test is meant to validate that the metadata is durable even when controller is replaced. Other tests in this class that use `killBroker` do not necessarily kill the controller, hence, I believe that there is value in keeping this test here.

I agree that we can change the name of the test. I have renamed it to `testMetadataDurabilityOnControllerFailure`.

Reply:
The value of the test still seems questionable to me. The call `waitForTopics` ensures metadata propagation to at least one broker. The call to `describeTopics` below after we have killed the controller is very likely to hit the same broker because the admin client biases toward open connections. A better test of metadata durability would probably kill all brokers and restart them. Anyway, we don't need to fix that here.